cubeviz/tools/collapse_cube.py

```diff
@@ -0,0 +1,359 @@
+            self.end_label.setText("End Wavelength:")
+
+        elif 'Custom' in newvalue and 'Indices' in newvalue:
+            # Custom indices
+            self.hide_start_end(False)
+            self.start_label.setText("Start Index:")
+            self.end_label.setText("End Index:")
+
+        else:
+            # Region defined in specviz
+            self.hide_start_end(True)
+
+            # We are going to store the start and end wavelengths in the text boxes even though
+            # they are hidden. This way we can use the text boxes later as a hidden storage container.
+            # TODO: Should probably save the ROIs so the start and end values are more accurate.
+            regex = r"-?(?:0|[1-9]\d*)(?:\.\d*)?(?:[eE][+\-]?\d+)?"
+            floating = re.findall(regex, newvalue)
+            self.start_text.setText(floating[0])
+            self.end_text.setText(floating[1])
+
+        # Let's update the text on the widget
+        if 'Custom' in newvalue:
+            self.widget_desc.setText(self._general_description + "\n\n" + self._custom_description)
+        else:
+            self.widget_desc.setText(self._general_description)
+
+    def hide_start_end(self, dohide):
+        """
+        Show or hide the start and end indices depending if the region
+        is defined from the specviz plot OR if we are using custom limits.
+
+        :param dohide:
+        :return:
+        """
+        if dohide:
+            self.start_label.hide()
+            self.start_text.hide()
+            self.end_label.hide()
+            self.end_text.hide()
+        else:
+            self.start_label.show()
+            self.start_text.show()
+            self.end_label.show()
+            self.end_text.show()
+
+    def calculate_callback(self):
+        """
+        Callback for when they hit calculate
+        :return:
+        """
+
+        # Grab the values of interest
+        data_name = self.data_combobox.currentText()
+        start_value = float(self.start_text.text().strip())
+        end_value = float(self.end_text.text().strip())
+
+        # Convert wavelengths to indices IF the region combo box is not the Indices selection
+        if not 'Indices' in self.region_combobox.currentText():
+            wavelengths = np.array(self.parent._wavelengths)
+            start_value = np.argsort(np.abs(wavelengths - start_value))[0]
+            end_value = np.argsort(np.abs(wavelengths - end_value))[0]
+
+        self.error_label_text.setText(' ')
+        self.error_label_text.setStyleSheet("color: rgba(255, 0, 0, 128)")
+
+        # Sanity checks first
+        if not start_value and not end_value:
+            self.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.end_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.error_label_text.setText('Must set at least one of start or end value')
+            return
+
+        if start_value and end_value and start_value > end_value:
```

Comment:
Just tried testing this by putting `1` in the start value box, and `0` in the end value box, and did not see any kind of error message.

Reply:
will be fixed in the next commit